browser/ai_rewriter/ai_rewriter_tab_helper.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_rewriter/ai_rewriter_tab_helper.h"
+
+#include <utility>
+
+#include "base/memory/weak_ptr.h"
+#include "brave/browser/ui/views/ai_rewriter/ai_rewriter_button_view.h"
```

Comment:
I think this is layer violation. using view outside of ui/views/. Can we avoid this?

Reply:
I think `browser()->window()->GetAIRewriterButton()` could work?